scripts/react_native_pods.rb

```diff
@@ -108,6 +108,11 @@ def use_react_native! (options={})
     pod 'libevent', '~> 2.1.12'
   end
 
+  if flipper_configuration.flipper_enabled && !production
```

Comment:
This is a temporary workaround that should fix https://github.com/facebook/react-native/issues/33764
If production is true the flipper pods are not added at all (differently than what CocoaPods does that compiles them, but then doesn't copy the frameworks), this avoids Flipper to be compiled when releasing, hence avoids the error discussed in the issue.
The downside of this workaround is that before making a release, the user needs to call `pod install` again

Reply:
> The downside of this workaround is that before making a release, the user needs to call pod install again

Correct me if I'm wrong, but it used to have to call it anyway, passing `production = false`, right?

I would also specify, perhaps also adding some doc, that it's still the case. When a user wants to build for release, it has to run `PRODUCTION=1 pod install`

Reply:
That is correct, however if there wasn't the mentioned issue we could only rely on CocoaPods' configurations logic to do esclude Flipper from production (it would anyway compile flipper but won't link it) and remove the production flag.
The `PRODUCTION=1 pod install` is how this is set up on RNTester, every final user could potentially set their own way to define if is production or not depending on their needs and their workflow.
If we think setting the `PRODUCTION` env variable is a good default to express that we want the production flag true I can add that to the template as well.

Reply:
Can you add a comment about that here? I think this can be easily missed since it's such a weird quirk of CocoaPods.

Reply:
Yes, sorry I missed this comment, will make another PR to fix those two comments as well

Reply:
Opened https://github.com/facebook/react-native/pull/33882